streams/src/main/java/org/apache/kafka/streams/state/internals/KeyValueSegments.java

```diff
@@ -35,7 +40,8 @@ public KeyValueSegment getOrCreateSegment(final long segmentId,
         if (segments.containsKey(segmentId)) {
             return segments.get(segmentId);
         } else {
-            final KeyValueSegment newSegment = new KeyValueSegment(segmentName(segmentId), name, segmentId);
+            final KeyValueSegment newSegment =
+                new KeyValueSegment(segmentName(segmentId), name, segmentId, metricsRecorder);
```

Comment:
The same recorder instance is passed to the RocksDB store instance of each segment.